hail/python/hail/dev-environment-py3.7.yml

```diff
@@ -0,0 +1,20 @@
+name: hail-py3.7
+dependencies:
+- python=3.7
+- Sphinx>=1.8.1
+- sphinxcontrib>=1.0
+- conda-forge::nbsphinx>=0.3.5
+- conda-forge::sphinx_rtd_theme>=0.4.2
+- numpy>=1.15.3
+- pandas>=0.23.4
+- matplotlib>=3.0.1
+- seaborn>=0.9.0
+- bokeh>=1.0.0
+- jupyter>=1.0.0
+- pip>=18.1
+- pytest>=3.9.2
+- pip:
+    - parsimonious>=0.8.1
+    - ipykernel>=4.9.0
+    - pytest-xdist>=1.22.2
+    - decorator>=4.3.0
```

Comment:
I guess we should talk about this, but I'm opposed to open future version compatibility. I think everything should be pinned to a major version (with flexibility on minor and patch). There's `pip` syntax for this (not sure about conda):
```
parsimonious==0.8.* # semver states each 0.x can break compatibility, not sure if parsimonious follows semver
ipykernel==4.*
```

Reply:
conda syntax is just `decorator=4.2`

Reply:
I think I'm against this.  I'm not sure what we gain.  How many times have we broken backward compatibility by accident?  I'm sure we're not the only ones.  It only leads to more differences we have to account for when doing difficult debugging.  I think we should pin versions hard and regularly update versions (e.g. quarterly).